backend/Application/Api/GraphQL/Network/NetworkQuery.cs

```diff
@@ -7,12 +7,33 @@ namespace Application.Api.GraphQL.Network;
 [ExtendObjectType(typeof(Query))]
 public class NetworkQuery
 {
+    private const int MaxSemverLength = 1024;
+
     [UsePaging]
-    public IEnumerable<NodeStatus> GetNodeStatuses([Service] NodeStatusSnapshot nodeSummarySnapshot)
+    public IEnumerable<NodeStatus> GetNodeStatuses(
+        [Service] NodeStatusSnapshot nodeSummarySnapshot,
+        NodeSortField sortField,
+        NodeSortDirection sortDirection)
     {
-        return nodeSummarySnapshot.NodeStatuses;
+        var statuses = nodeSummarySnapshot.NodeStatuses.AsQueryable();
+        statuses = sortField switch
+        {
+            NodeSortField.AveragePing => sortDirection == NodeSortDirection.ASC ? statuses.OrderBy(s => s.AveragePing) : statuses.OrderByDescending(s => s.AveragePing),
+            NodeSortField.BlocksReceivedCount => sortDirection == NodeSortDirection.ASC ? statuses.OrderBy(s => s.BlocksReceivedCount) : statuses.OrderByDescending(s => s.BlocksReceivedCount),
+            NodeSortField.ClientVersion => sortDirection == NodeSortDirection.ASC
+                ? statuses.OrderBy(s => Semver.SemVersion.Parse(s.ClientVersion, Semver.SemVersionStyles.Any, MaxSemverLength), Semver.SemVersion.SortOrderComparer)
+                : statuses.OrderByDescending(s => Semver.SemVersion.Parse(s.ClientVersion, Semver.SemVersionStyles.Any, MaxSemverLength), Semver.SemVersion.SortOrderComparer),
+            NodeSortField.ConsensusBakerId => sortDirection == NodeSortDirection.ASC ? statuses.OrderBy(s => s.ConsensusBakerId) : statuses.OrderByDescending(s => s.ConsensusBakerId),
+            NodeSortField.FinalizedBlockHeight => sortDirection == NodeSortDirection.ASC ? statuses.OrderBy(s => s.FinalizedBlockHeight) : statuses.OrderByDescending(s => s.FinalizedBlockHeight),
+            NodeSortField.NodeName => sortDirection == NodeSortDirection.ASC ? statuses.OrderBy(s => s.NodeName) : statuses.OrderByDescending(s => s.NodeName),
+            NodeSortField.PeersCount => sortDirection == NodeSortDirection.ASC ? statuses.OrderBy(s => s.PeersCount) : statuses.OrderByDescending(s => s.PeersCount),
+            NodeSortField.Uptime => sortDirection == NodeSortDirection.ASC ? statuses.OrderBy(s => s.Uptime) : statuses.OrderByDescending(s => s.Uptime),
+            _ => throw new NotImplementedException()
```

Comment:
Can this happen? Why not just do nothing in that case instead of raising an exception?

Reply:
This can happen when a new field NodeSortField is added without it being handeled here for sorting. 
So if that field is selected by the user (or developer who is adding it). A `NotImplementedException` will be better than doing nothing.

Reply:
Ok.